grammar.peg

```diff
@@ -105,15 +105,29 @@ MatchNotMatches <- _ "not" _ "matches" _ {
 
 Selector "selector" <- first:Identifier rest:SelectorOrIndex* {
    sel := Selector{
-      first.(string),
+      Type: SelectorTypeBexpr,
+      Path: []string{first.(string)},
    }
-
    if rest != nil {
       for _, v := range rest.([]interface{}) {
-        sel = append(sel, v.(string))
+        sel.Path = append(sel.Path, v.(string))
       }
    }
    return sel, nil
+} / '"' ptrsegs:JsonPointerSegment* '"' {
+   sel := Selector{
+      Type: SelectorTypeJsonPointer,
+   }
+   if ptrsegs != nil {
+      for _, v := range ptrsegs.([]interface{}) {
+         sel.Path = append(sel.Path, v.(string))
+      }
+   }
+   return sel, nil
+}
+
+JsonPointerSegment <- '/' ident:[\pL\pN-_.~|]+ {
+   return string(c.text)[1:], nil
```

Comment:
Should we do `~0` > `~` and `~1` -> `/` expansion here?

Reply:
Nope. If we do, the constructed path doesn't actually match the input. If we were to then e.g. put it back together into a string via strings.Join, we'd end up with a different string. We should do it either:

1) Never, if we reconstruct the string and pass it directly to pointerstructure, since it will then do the substitution; or
2) Before calling pointerstructure if we are setting its path directly based on our parsed path

Reply:
:thumbsup: